build.sbt

```diff
@@ -15,52 +18,65 @@ ThisBuild / libraryDependencySchemes ++= Seq(
   "com.github.plokhotnyuk.jsoniter-scala" % "jsoniter-scala-core" % "early-semver"
 )
 
-ThisBuild / versionPolicyIntention := Compatibility.BinaryCompatible
+ThisBuild / versionPolicyIntention := Compatibility.None
```

Comment:
No binary compat?

Reply:
The main reason is this:

```scala
trait ScalarTag[@spec(Boolean, Int, Long, Float, Double) T]
    extends ClassTag[T] { .. }

type CLM[C] = ClassTag[C]

class ScalarTagAny[T: CLM] extends ScalarTag[T]
```
ScalarTagAny is recursive because itself is a ClassTag. This was working in scala2 because stAny has no type annotation thereby is not eligible for recursion in:

```
implicit def stAny[T: CLM] = new ScalarTagAny[T] { ... }
```
. If we place the type annotation then it blows up in runtime (scala3 needs that). I did not try it with an implicit class but that can't be top level and ScalarTagAny is top level in the current main branch.

I reworked that part by making ScalarTag compose with ClassTag and not inherit it. 
Loosely connected to this I found some parts which I deemed hard to maintain (scalarTagAnyVal vs scalaTagAny vs scalarTagAnyRef and the takeType in Frame) so I removed those.

I did not remove but unless the implicit conversions for `na` can be fixed that is also a candidate for removal. It is very confusing that there is an `NA` and an `na` object.

Reply:
> I did not remove but unless the implicit conversions for na can be fixed that is also a candidate for removal. It is very confusing that there is an NA and an na object.

Agreed. Also the fact that `na` plays poorly with type inference. For example, `Vec(1, na)` doens't infer as `Vec[Int](1, na)`.

> `class ScalarTagAny[T: CLM] extends ScalarTag[T]`

Oho, didn't notice this unnecessary composition and inheritance pattern. Nice catch.

Reply:
Addressed in 58b38da.